src/Features/CSharp/Portable/CodeRefactorings/ConvertToRecord/CSharpConvertToRecordRefactoringProvider.cs

```diff
@@ -0,0 +1,791 @@
+                return false;
+            }
+
+            if (property.Initializer != null || property.ExpressionBody != null)
+            {
+                return false;
+            }
+
+            var propAccessibility = CSharpAccessibilityFacts.Instance.GetAccessibility(member);
+
+            // more restrictive than internal (protected, private, private protected, or unspecified (private by default))
+            if (propAccessibility < Accessibility.Internal)
+            {
+                return false;
+            }
+
+            // no accessors declared
+            if (property.AccessorList == null)
+            {
+                return false;
+            }
+
+            // for class records and readonly struct records, properties should be get; init; only
+            // for non-readonly structs, they should be get; set;
+            // this is bacause those are the defaults for properties generated from positional constructors
+            // for each respective class type, although we could decide to change this requirement
+            // the get should be public
+            // neither should have bodies (as it indicates more complex functionality)
+            var accessors = property.AccessorList.Accessors;
+            if (accessors.Any(a => a.Body != null || a.ExpressionBody != null) ||
+                !accessors.Any(a => a.Kind() == SyntaxKind.GetAccessorDeclaration && a.Modifiers.IsEmpty()))
+            {
+                return false;
+            }
+
+            if (containingType.TypeKind == TypeKind.Struct && !containingType.IsReadOnly)
+            {
+                if (!accessors.Any(a => a.IsKind(SyntaxKind.SetAccessorDeclaration)))
+                {
+                    return false;
+                }
+            }
+            else
+            {
+                // either we are a class or readonly struct
+                if (!accessors.Any(a => a.IsKind(SyntaxKind.InitAccessorDeclaration)))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private static SyntaxList<AttributeListSyntax> GetModifiedAttributeListsForProperty(PropertyDeclarationSyntax p)
+            => SyntaxFactory.List(p.AttributeLists.SelectAsArray(attributeList =>
+            {
+                if (attributeList.Target == null)
+                {
+                    // convert attributes attached to the property with no target into "property :" targeted attributes
+                    return attributeList
+                        .WithTarget(SyntaxFactory.AttributeTargetSpecifier(SyntaxFactory.Token(SyntaxKind.PropertyKeyword)))
+                        .WithoutTrivia();
+                }
+                else
+                {
+                    return attributeList.WithoutTrivia();
+                }
+            }));
+
+        private static ImmutableArray<MemberDeclarationSyntax> GetModifiedMembersForPositionalRecord(
```

Comment:
consider doc'ing what thsi does.  specifically what you mean by 'modified members'.